### PR TITLE
chore: restructure action dropdown component

### PR DESCRIPTION
**Pull request by hamidzr**
## Description

prepare ActionDropdown, Icon, and Spinner to be shared with saas

## Commentary (optional)

general steps with these prs
```
## process

- move the desired components, snippets, and assets into a directory eg `tmp-shared` other than shared.
we do this so that this can still be easily tracked with git while you're working on it as well is for the reviewer to see the changes easily.

- (review step 1) ask for a review at this point

- make a final move from this temporary directory to `src/shared`.
create a separate branch with your changes for the shared-web repo eg `411-share-utils`
- let ci run tests against it through your pr. check that all tests are passing

- (review step 2, instead of step 1) ask for a review at this point

- create a pr in saas to merge the branch into main
  -  alternatively: squash merge `git merge --squash origin/4112-share-utils` ()
- update the original pr's submodule to point to the updated commit sha and land your changes



prs and commits used in this example:
core pr: https://github.com/determined-ai/determined/pull/4112
shared-web commit: 1114dc1aeb570a6574e3049f9f7b2ebc37647d1f


### what does this solve?

- avoid changing shared-web/main before your changes to the shared code is tested and reviewed in saas and core.
- helping the reviewer, and yourself, see the changes in git without having to go into another submodule.



```

## Checklist

- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
See [Release Note](../docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

[DET-1234]: https://determinedai.atlassian.net/browse/DET-1234?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by hamidzr** [comment]
I think this is a good spot to review since you can see all the changes. once it's approved I'll 
- merge `tmp-shared` with `shared` which would be in a separate repo
- land the changes in shared-web

File: webui/react/src/tmp-shared/components/Icon/index.ts [hunk 1]
```diff
@@ -0,0 +1,3 @@
+export * from './Icon';
+import Icon from './Icon';
+export default Icon;
```
Comment (trentwatt) [inline webui/react/src/tmp-shared/components/Icon/index.ts:3]: question: Why are these needed? Will we have to do this with every component?

**Review by trentwatt — APPROVED** [review]
Looks fine, but I guess the implications of the submodule approach are just now dawning on me, and I'm a bit terrified.
